src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdkAppEngineDeployment.java

```diff
@@ -32,19 +32,26 @@
  */
 public class CloudSdkAppEngineDeployment implements AppEngineDeployment {
 
-  private CloudSdk sdk;
+  private final CloudSdk sdk;
 
-  public CloudSdkAppEngineDeployment(
-      CloudSdk sdk) {
-    this.sdk = sdk;
+  public CloudSdkAppEngineDeployment(CloudSdk sdk) {
+    this.sdk = Preconditions.checkNotNull(sdk);
   }
 
+  /**
+   * Deploys a project to App Engine.
+   *
+   * @throws CloudSdkNotFoundException when the Cloud SDK is not installed where expected
+   * @throws CloudSdkOutOfDateException when the installed Cloud SDK is too old 
+   * @throws AppEngineException when there is an issue transmitting the data to the server
```

Comment:
@throws AppEngineException when there is an issue uploading project files to the cloud